include/pmacc/math/functions/Log.hpp

```diff
@@ -22,29 +22,20 @@
 
 #pragma once
 
-#include "pmacc/types.hpp"
+#pragma once
 
-#include <cmath>
+#include "pmacc/math/functions/Common.hpp"
 
+#include <alpaka/alpaka.hpp>
 
-namespace pmacc
+namespace pmacc::math
 {
-    namespace math
-    {
-        template<>
-        struct Log10<float>
-        {
-            using result = float;
-
-            HDINLINE float operator()(const float& value)
-            {
-#if(CUPLA_DEVICE_COMPILE == 1) // we are on gpu
-                return ::log10f(value);
-#else
-                return ::log10(value);
-#endif
-            }
-        };
-
-    } // namespace math
-} // namespace pmacc
+    //! Computes the natural (base e) logarithm.
+    ALPAKA_UNARY_MATH_FN(log, alpaka::math::ConceptMathLog, Log)
+
+    //! Computes the natural (base 2) logarithm.
```

Comment:
```suggestion
    //! Computes the logarithm to the base of 2.
```
natural logarithm is only to base e